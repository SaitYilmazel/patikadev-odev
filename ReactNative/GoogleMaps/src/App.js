import React, {useState, useRef} from 'react';
import {View, StyleSheet} from 'react-native';
import MapView from 'react-native-maps';
import Config from 'react-native-config';

import {InfoCard, Loading, UserMarker} from './components';
import useFetch from './hooks/useFetch';

export default function App() {
  const [user, setUser] = useState();
  const [infoModalVisibility, setInfoModalVisibility] = useState();
  const {data, loading, error} = useFetch(Config.API_URL);
  const mapRef = useRef();

  const renderUserMarker = () => {
    return data.map(
      ({
        id,
        first_name,
        last_name,
        email,
        gender,
        phone_number,
        avatar,
        address: {coordinates, country, city, street_name, street_address},
      }) => {
        return (
          <UserMarker
            key={id}
            userImageURL={avatar}
            coordinates={{
              latitude: coordinates.lat,
              longitude: coordinates.lng,
            }}
            onSelect={() => {
              handleMarkerSelect(coordinates, {
                avatar,
                first_name,
                last_name,
                email,
                gender,
                phone_number,
                country,
                city,
                street_name,
                street_address,
              });
            }}
          />
        );
      },
    );
  };

  function handleMarkerSelect(coor, selectedUser) {
    setUser(selectedUser);
    handleModalVisibility();
    mapRef.current.animateToRegion({
      latitude: coor.lat,
      longitude: coor.lng,
      latitudeDelta: 7,
      longitudeDelta: 7,
    });
  }

  function handleModalVisibility() {
    setInfoModalVisibility(!infoModalVisibility);
  }

  return (
    <View style={styles.container}>
      <MapView ref={mapRef} style={styles.map}>
        {data && renderUserMarker()}
      </MapView>
      {loading && <Loading />}
      {user && (
        <InfoCard
          visible={infoModalVisibility}
          onClose={handleModalVisibility}
          user={user}
        />
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  map: {
    flex: 1,
  },
});
