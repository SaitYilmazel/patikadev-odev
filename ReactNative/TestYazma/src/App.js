import React, {useState} from 'react';
import {View, Text, TextInput, FlatList, StyleSheet} from 'react-native';

import {Button} from './components';

export default function App() {
  const [text, setText] = useState('');
  const [list, setList] = useState([]);

  function addToList() {
    if (!text) {
      return;
    }
    setList([...list, text]);
  }

  const renderElements = ({item}) => <Text>{item}</Text>;
  return (
    <View style={styles.container}>
      <FlatList
        testID="list"
        data={list}
        keyExtractor={(_, index) => index.toString()}
        renderItem={renderElements}
      />
      <TextInput
        testID="input"
        style={styles.input}
        placeholder="Add..."
        onChangeText={setText}
      />
      <Button title="Add" onPress={addToList} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  input: {
    backgroundColor: '#e0e0e0',
    margin: 10,
    padding: 10,
    borderRadius: 10,
  },
});
