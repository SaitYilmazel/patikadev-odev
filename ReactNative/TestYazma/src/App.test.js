import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';

import App from './App';

test('should match with snapshot', () => {
  const comp = render(<App />);
  expect(comp).toMatchSnapshot();
});

test('should add Ä±tem to list', () => {
  const comp = render(<App />);
  const textInput = comp.getByTestId('input');
  fireEvent.changeText(textInput, 'test element');

  const submitButton = comp.getByTestId('button-touchable');
  fireEvent.press(submitButton);

  const list = comp.getByTestId('list').props;

  expect(list.data.length).toBe(1);
});


test('should not add item if input is empty', () => {
  const comp = render(<App />);

  const submitButton = comp.getByTestId('button-touchable');
  fireEvent.press(submitButton);

  const list = comp.getByTestId('list').props;
  expect(list.data.length).toBe(0);
});