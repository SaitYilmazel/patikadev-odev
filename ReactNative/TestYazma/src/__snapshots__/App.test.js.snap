// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="list"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
  <TextInput
    onChangeText={[Function]}
    placeholder="Add..."
    style={
      Object {
        "backgroundColor": "#e0e0e0",
        "borderRadius": 10,
        "margin": 10,
        "padding": 10,
      }
    }
    testID="input"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "dodgerblue",
        "borderRadius": 10,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="button-touchable"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "bold",
        }
      }
      testID="button-title"
    >
      Add
    </Text>
  </View>
</View>
`;
